### ISSUE TITLE
Namespace/prefix generated aliases with `$` or similar

### ISSUE THREAD

**Issue by jasonwoodland** [acarl005/join-monster#532]
### Description

Aliases generated with setting `minify` to true can generate aliases as simple as "a" or "b". These single-letter aliases are commonly used in SQL subqueries. We can define subqueries in join-monster's where and join expressions.

To prevent clashing with join-monster's generated aliases in our subqueries, I propose that join-monster prefix the aliases generated with a "$" or similar less commonly used character. I.e. "a" would become "$a".

If this is too intrusive it could be enabled through a config parameter such as `prefixAliases: true` or `aliasNamespace: "$"`

Example subquery used in a join-monster where expression:
```sql
(
  SELECT p.first_name || ' ' || p.last_name
  FROM agent a -- using alias 'a'
  INNER JOIN person p ON a.person_id = p.id
  WHERE agent.id = ${p}.agent_id -- ${p} may possibly be 'a' which would cause a clash
  LIMIT 1
)
```

### Reproduction

N/a

### Environment

N/a

**Comment by nicoabie** [comment]
Interesting!

Can you show with a https://github.com/join-monster/join-monster-sscce that this actually happens?

Becase in the method that you can write raw sql, you have a variable with what will be the name of the table. So I'm not sure which kind of query could break that.

**Comment by jasonwoodland** [comment]
Thanks for the quick reply.

Yes, we can write raw SQL, but we don’t know what alias join-monster will pass to our join/where expression function.

If we use the alias ‘a’ in our expression, it could clash with the ‘a’ that join-monster assigns to its first table reference. This conflict arises because join-monster assigns aliases automatically, and we might unknowingly use the same alias.

Since it’s common to use single letters in SQL subqueries, I’d prefer if join-monster avoided using single-letter aliases. Instead, it would be better if join-monster prefixed its generated aliases with a ‘$’ to prevent any clashes with our own aliases.

See
https://github.com/join-monster/join-monster/blob/master/src/alias-namespace.js

I would do something as trivial as:
```diff
  generate(type, name) {
    // if minifiying, make everything ugly and unique.
    if (this.minify) {
      // tables definitely all need unique names
      if (type === 'table') {
-       return this.mininym.next().value.join('')
+       return `$${this.mininym.next().value.join('')}`;
      }

      // but if its a column, we dont need to worry about the uniqueness from other columns
      // because the columns will get prefixed with the parent(s)
      if (!this.columnAssignments[name]) {
        this.columnAssignments[name] = this.mininym.next().value.join('')
      }

      return this.columnAssignments[name]
    }
  }
```

Edit:

Here is an excerpt from the SQL query join-monster generated with `minify: true`:
```sql
SELECT
  ...
  "az"."id" AS "ap__az__b",
  "az"."name" AS "ap__az__h",
  "az"."color" AS "ap__az__ao"
FROM (
  SELECT "a".*, count(*) OVER () AS "$total"
  FROM property_ext "a" -- <- you can see join-monster has assigned property_ext to "a"
  WHERE ("a".deleted_at IS NULL AND lower((
          SELECT p.first_name || ' ' || p.last_name
          FROM agent AS a -- but I have also used "a" in my WHERE subquery
          INNER JOIN person AS p ON a.person_id = p.id
          WHERE a.id = "a".agent_id -- causing a clash on this subquery WHERE expression
          LIMIT 1
        )) LIKE '%' || lower('a') || '%' AND "a".id NOT IN (
...
```

**Comment by nicoabie** [comment]
Yeah, now I see it clearly!

If you want to submit a PR we can include this in the next release.

I would by default have no prefix so we don't introduce any breaking changes and add an option to opt in for a prefix.

**Comment by jasonwoodland** [comment]
Thank you @nicoabie! PR ready for review.

**Comment by nicoabie** [comment]
FIxed in #533
